### PR TITLE
[WIP] Setup.py

### LINKED ISSUE #413: Create setup.py
In line with the decisions taken for the HardwareRepository project about the _setup.py_ file, I think it would be also interesting to create a _setup.py_ file for the mxcube/mxcube project (the qt front-end version) with the aim to facilitate the deployment of the application. What do you think about it?

### PR DESCRIPTION
**Pull request by jordiandreu**
This changes are though to be on top of branch clear_create_app (#409):

* Create setup.py file
* Rename folder gui -> mxcubeqt: the new package, name is just an example and need to be renamed in common agreement..
* Rename folder bin -> scripts: additional scripts used by the project but not included in the module.
    ** bin/mxcube renamed to mxcubeqt/apps/mxcube.py
* Rename folder configuration -> mxcubeqt/examples: only the mockup GUI yaml file.
    ** The fakeimg.jpg is proposed to be moved to HardwareRepository
    ** Remove site specific files.
* Create run method in mxcubeqt/apps/mxcube.py and create entrypoint named mxcube.
* Added optional flag -v/--version.

### CODE REVIEW

**Comment by jordiandreu** [comment]
### Additional notes:

* Few changes are required also in the HardwareRepository side (mxcube-417).

* Use pkg_resources avoiding to hardcode paths. For instance, loading the _fakeimg.jpg_ file in _QtVideoMockup.py_ has been proposed in the HWR branch.

* Current version also deploys the HardwareRepository, which will be removed from the project later (some modification will be required).

* Regarding the _setup.py_ file, some information is incomplete or not correct and need some review:
    * author name
    * author email
    * classifiers
    * requirements

* I have included the bumversion configuration (a new _-v / --version_ flag also to check the version from the launcher).

* Pending: Add MANIFEST.in file to create tar distributions.


**How to test it?**
I have also included a conda environment recipe in scripts folder, which will provide all dependencies required to run the mockup version of the application. To install the module into the conda environment (once created):

```
conda activate mxcube-py36
cd mxcube
pip install --no-deps

```
After the installation, a mxcube launcher should be available to run the application.

Cheers

File: mxcubeqt/BaseComponents.py [hunk 1]
```diff
@@ -203,7 +203,7 @@ def set_run_mode(mode):
                         widget.set_expert_mode(False)
                     except BaseException:
                         logging.getLogger().exception(
-                            "Could not set %s to user mode", widget.name()
+                            "Could not set %s to user mode", 'rrr'#widget.name#()
```
Comment (IvarsKarpics) [inline mxcubeqt/BaseComponents.py]: ?

File: mxcubeqt/bricks/ALBA/ALBA_ActuatorBrick.py [hunk 2]
```diff
@@ -19,8 +19,8 @@
 
 import logging
 
-from gui.utils import Colors, QtImport
-from gui.BaseComponents import BaseWidget
+from mxcubeqt.utils import Colors, QtImport
+from mxcubeqt.BaseComponents import BaseWidget
```
Comment (IvarsKarpics) [inline mxcubeqt/bricks/ALBA/ALBA_ActuatorBrick.py:23]: base_components

Comment (IvarsKarpics) [reply mxcubeqt/bricks/ALBA/ALBA_ActuatorBrick.py:23]: I guess we have to change all file names to snake_case (colors, qt_import, etc). I started it in pr #409

File: mxcubeqt/BaseComponents.py (hunk 1 above)
Comment (jordiandreu) [reply mxcubeqt/BaseComponents.py]: Sorry, I forgot to remove this dummy trace...Fixed it!

File: mxcubeqt/bricks/ALBA/ALBA_ActuatorBrick.py (hunk 2 above)
Comment (jordiandreu) [reply mxcubeqt/bricks/ALBA/ALBA_ActuatorBrick.py:23]: Indeed. I was only using your proposal to implement the setup.py draft. I agree with #409 and keep renaming in the same direction.

**Comment by IvarsKarpics** [comment]
Hi @jordiandreu ,

Thank you for this massive work. I will not be able to review it fully and will come back to it after the winter break. At first we have to merge the PRs in the HardwareRepository and then we can work on this one.

**Comment by IvarsKarpics** [comment]
Hi @jordiandreu  and @rhfogh 

I would like to merge this PR and proceed with testing.  Are you fine with the changes?

**Review by rhfogh — COMMENTED** [review]
I have not actually looked (478 files! - and I am not up on setup.py). But I'll take your word for it and approve unseen.

**Pull request merged**